casanovo/denovo/model_runner.py

```
@@ -670,6 +678,17 @@ def _get_strategy(self) -> Union[str, DDPStrategy]:
             return "auto"
 
+    def _validate_vocab_compatibility(self, tokenizer) -> None:
+        model_vocab_size = self.model.vocab_size
+        tokenizer_vocab_size = len(tokenizer) + 1
+        if model_vocab_size > tokenizer_vocab_size:
+            raise ValueError(
+                "The model was trained with a vocabulary of size "
+                f"{model_vocab_size}, but the current config only defines "
+                f"a vocabulary of size {tokenizer_vocab_size}. Ensure your "
+                "config's `residues` match those used during training."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)